demos/sample_azure_iot_ca_recovery/sample_azure_iot_ca_recovery.c

```diff
@@ -330,8 +403,35 @@ static void prvAzureDemoTask( void * pvParameters )
                                            &pulIothubHostnameLength, &pucIotHubDeviceId,
                                            &pulIothubDeviceIdLength ) ) != 0 )
         {
-            LogError( ( "Failed on sample_dps_entry!: error code = 0x%08x\r\n", ulStatus ) );
-            return;
+            if( ulStatus == sampleazureiotRECOVERY_INITIATED )
+            {
+                memset( &xNetworkCredentials, 0, sizeof( xNetworkCredentials ) );
+                ulStatus = prvSetupRecoveryNetworkCredentials( &xNetworkCredentials );
+                configASSERT( ulStatus == 0 );
```

Comment:
Same here - it looks like other errors print a log and return

Reply:
The returns should debatably be asserts since you shouldn't return from a task function. I'll put as assert since they're pretty critical and most functions called inside of them are "assert"ed